server/server.go

```diff
@@ -400,7 +400,7 @@ func (s *Server) initRouter() {
 	s.Handler = router
 }
 
-func (s *Server) execQuery(ctx context.Context, r *http.Request, query string, input ast.Value, explainMode types.ExplainModeV1, includeMetrics, includeInstrumentation, pretty bool) (results types.QueryResponseV1, err error) {
+func (s *Server) execQuery(ctx context.Context, r *http.Request, query string, input ast.Value, explainMode types.ExplainModeV1, includeMetrics, includeInstrumentation, pretty bool, parsedQuery ast.Body) (results types.QueryResponseV1, err error) {
```

Comment:
Let's replace `query string` with `query ast.Body`. The string version is not required anymore.

Reply:
there are log function down in the call path that take query string and not ast.Body. Those would all need to change?

Reply:
It should be fine to call `String()` on the body. I'd prefer that to having two arguments that specify the same thing.